ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/ad_utils.py

```diff
@@ -15,6 +19,12 @@
 
 
 class AdUtils:
+    terminal_job_states = [DetectAnomalyJob.LIFECYCLE_STATE_CANCELED,
+                           DetectAnomalyJob.LIFECYCLE_STATE_FAILED,
+                           "PARTIALLY_SUCCEEDED",
```

Comment:
Please use the constants provided by the SDK (like `DetectAnomalyJob.LIFECYCLE_STATE_FAILED` in the previous line) and not have hard-coded values to avoid any expected behavior.